main.py

```diff
@@ -16,24 +16,25 @@
 screen.listen()
 screen.onkey(player.turtle_move_up, "Up")
 screen.onkey(player.turtle_move_down, "Down")
-car_manager = Car()
+car_manager = CarManager()
+car_manager.create_cars()
 game_on = True
 
 while game_on:
     scoreboard.write(f"Score: {score}", font=("Arial", 24, "bold"))
-    car_manager.create_car()
-    car_manager.car_move()
+    car_manager.move_cars()
     screen.update()
     time.sleep(0.1)
-    for car in car_manager.all_cars:
+    car_manager.traffic_handler()
+    for car in car_manager.car_list:
         if car.distance(player) < 30:

```

Comment:
You could move this logic to the `CarManager`. For example, a function `check_player_collision` that receives a `Player` as a parameter.